domain/src/test/java/org/oppia/domain/profile/ProfileManagementControllerTest.kt

```diff
@@ -95,8 +122,392 @@ class ProfileManagementControllerTest {
 
   @Test
   @ExperimentalCoroutinesApi
-  fun testProfileManagementController() = runBlockingTest(coroutineContext) {
-    // TODO(#16): Finish test cases with full implementation
+  fun testAddProfile_addProfile_checkProfileIsAdded() = runBlockingTest(coroutineContext) {
+    profileManagementController.addProfile("James", "123", null, true).observeForever(mockUpdateResultObserver)
+    advanceUntilIdle()
+
+    val profileDatabase = ProfileDatabase.parseFrom(FileInputStream(File(context.filesDir, "profile_database.cache")))
+
+    verify(mockUpdateResultObserver, atLeastOnce()).onChanged(updateResultCaptor.capture())
+    assertThat(updateResultCaptor.value.isSuccess()).isTrue()
+    val profile = profileDatabase.profilesMap[0]
+    assertThat(profile?.name).isEqualTo("James")
+    assertThat(profile?.pin).isEqualTo("123")
+    assertThat(profile?.allowDownloadAccess).isEqualTo(true)
+    assertThat(profile?.id?.internalId).isEqualTo(0)
```

Comment:
Won't id always be present since it's a proto?

Reply:
Removed null check.